aten/src/ATen/native/mkldnn/Pooling.cpp

```diff
@@ -73,29 +73,51 @@ static Tensor _mkldnn_pool2d(
     IntArrayRef dilation,
     bool ceil_mode,
     ideep::algorithm algo) {
-  TORCH_CHECK(!ceil_mode, "Currently Mkldnn Pooling operators do not support ceil_mode.");
   auto kernel_size_vec = expand_param_if_needed(kernel_size, "kernel_size", 2);
   auto stride_vec = expand_param_if_needed(stride, "stride", 2);
   auto padding_vec = expand_param_if_needed(padding, "padding", 2);
+  auto padding_vec_l = padding_vec;
+  auto padding_vec_r = padding_vec;
   auto dilation_vec = expand_param_if_needed(dilation, "dilation", 2);
 
   const ideep::tensor& x = itensor_from_mkldnn(input);
-  const std::vector<int64_t> output_sizes = pool_output_sizes(
+  std::vector<int64_t> output_sizes = pool_output_sizes(
       input.sizes(),
       kernel_size_vec,
       stride_vec,
       padding_vec,
       dilation_vec,
-      ceil_mode);
+      false /*ceil_mode */);
+
+  if (ceil_mode) {
+    // MKLDNN does not support ceil mode, so we adjust padding
+    // on the right side to match behavior. Adjust output size
+    // accordingly.
+    const std::vector<int64_t> output_sizes_ceil = pool_output_sizes(
+      input.sizes(),
+      kernel_size_vec,
+      stride_vec,
+      padding_vec,
+      dilation_vec,
+      true // ceil_mode
+   );
+
+    for (unsigned i = 2; i < input.sizes().size(); ++i) {
+     TORCH_CHECK(output_sizes_ceil[i] >= output_sizes[i]);
+     padding_vec_r[i - 2] += output_sizes_ceil[i] - output_sizes[i];
```

Comment:
I think this is incorrect, it should be
```padding_vec_r[i - 2] = (output_sizes_ceil[i] - 1) * strides_vec[i - 2] + kernel_size_vec[i] - input.sizes(i) - padding_vec_l[i - 2]```

The diff between output_sizes_ceil[i] and output_sizes[i] is `<=1` while there are cases we need to pad more more than one pixel.

Reply:
we don't need to compute output_sizes_ceil again there? it will compute twice output_sizes for ceil case.

Reply:
Ah, yes i got confused...